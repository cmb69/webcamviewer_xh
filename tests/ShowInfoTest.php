<?php

/**
 * Copyright 2012-2022 Christoph M. Becker
 *
 * This file is part of Webcamviewer_XH.
 *
 * Webcamviewer_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Webcamviewer_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Webcamviewer_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Webcamviewer;

use function XH_includeVar;
use PHPUnit\Framework\TestCase;
use ApprovalTests\Approvals;

class ShowInfoTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $systemCheckerStub = $this->createStub(SystemChecker::class);
        $systemCheckerStub->method('checkVersion')->willReturn(true);
        $systemCheckerStub->method('checkExtension')->willReturn(true);
        $systemCheckerStub->method('checkWritability')->willReturn(true);
        $subject = new ShowInfo(
            "./",
            $systemCheckerStub,
            XH_includeVar("./languages/en.php", "plugin_tx")['webcamviewer']
        );

        $response = $subject();

        Approvals::verifyString($response);
    }
}
