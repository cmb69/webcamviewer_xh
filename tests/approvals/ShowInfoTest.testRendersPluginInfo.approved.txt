Webcamviewer\Response Object
(
    [output:Webcamviewer\Response:private] => 
<h1>Webcamviewer 2.0-dev</h1>
<h2>System check</h2>
<p class="xh_success">PHP version ≥ 7.0.0 ⇒ okay</p>
<p class="xh_success">Extension 'json' loaded ⇒ okay</p>
<p class="xh_success">CMSimple_XH version ≥ 1.7.0 ⇒ okay</p>
<p class="xh_success">Folder './config/' writable ⇒ okay</p>
<p class="xh_success">Folder './languages/' writable ⇒ okay</p>

    [hjs:Webcamviewer\Response:private] => 
)
